### ISSUE TITLE
Hard-coded default value for manifestVersionRegularExpression / currentVersionRegex

### ISSUE THREAD

**Issue by pfink** [openhab/static-code-analysis#304]
We have a hard-coded value for manifestVersionRegularExpression / currentVersionRegex [here](https://github.com/openhab/static-code-analysis/blob/master/sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml#L150).

This is a very bad smell as it will immediately break the code style check after a version bump. @VelinYordanov @svilenvul: Any suggestion for resolution? Or am I wrong?

**Comment by svilenvul** [comment]
Have you tried to add a new property to a checkstyle.properties file like this one - https://github.com/openhab/openhab-core/blob/master/tools/checkstyle.properties :

```
checkstyle.headerCheck.content=^/\\*\\*$\\n^ \\* Copyright \\(c\\) {0}-{1} by the respective copyright holders\\.$\\n^ \\*$\\n^ \\* All rights reserved\\. This program and the accompanying materials$\\n^ \\* are made available under the terms of the Eclipse Public License v1\\.0$\\n^ \\* which accompanies this distribution, and is available at$\\n^ \\* http://www.eclipse.org/legal/epl\\-v10\\.html$
checkstyle.headerCheck.values=2015,2018 
checkstyle.pomXmlCheck.currentVersionRegex= add_your_regex_here
```

**Comment by dstivanov** [comment]
@pfink The regex is taken from `${checkstyle.pomXmlCheck.currentVersionRegex}` and as it is always needed with added a default value 2.3.0. So the suggestion of @svilenvul should work well. 
Please do not hesitate to contact us again if you have other problems.

**Comment by pfink** [comment]
Yes, I know this solution and [I already use it for my binding](https://github.com/pfink/openhab2-flicbutton/blob/master/tools/checkstyle.properties#L3). But it only moves the problem to another location, it is still a hard-coded value that needs extra maintenance on every version bump. Currently, it seems to be undocumented that this value has to be changed on every release and it could be easily forgetten. Also, the value is not overriden on our official repos (e.g. [here](https://github.com/openhab/openhab2-addons/blob/master/tools/checkstyle.properties)). So for them, currently the default value is used which means:
- we always have to create a static-code-analysis release for each openHAB GA release (every time when there is a snapshot version bump)
- we should integrate this version bump into our automated openHAB release process (otherwise it can be easily forgotten)
OR
- we find a better solution where this version is dynamically determined from the POM version

I think the latter version would be more smart & flexible. @kaikreuzer: wdyt?

**Comment by pfink** [comment]
@kaikreuzer: Why closed?

**Comment by kaikreuzer** [comment]
Because https://github.com/openhab/static-code-analysis/pull/306 has been (manually) merged and is thus part of 0.6.0.

**Comment by pfink** [comment]
Now I see. Thanks!
